course6/toyapp_mikhail/displaylib/vector_common.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+namespace Display{
+
+template <class TValue, int TDimentions>
+class VectorCommon
+{
+	// No constructor for vector without specification
+	VectorCommon() = delete;
+};
+
+// Specification for 2 dimentional vector
+template <class TValue>
```

Comment:
no no no, this is not what I had in mind. The goal of this template should be to use vector of any size/type

Reply:
Changed.
But I still want to use nice decorations with x,y and x,y,z to not change client code.
Using v.x, v.y in client code is cleaner than v[0], v[1].
For decorations I had to use inheritance. is it ok?

Reply:
Forget about ```v.x```, ```v.y```. C++ doesn't have the support for properties.
The canonical way is to use accessor methods: getters and setters.

You may implement ```v.x()```, ```v.y()``` in an inherited class, and they can be an alternative api to ```v[0]```.

You will have to change your client code one way or the other. Consider this a punishment for using public fields :)

Reply:
Ok